spring-cloud-dataflow-module-deployers/spring-cloud-dataflow-module-deployer-lattice/src/main/java/org/springframework/cloud/dataflow/module/deployer/lattice/LrpModuleDeployer.java

```diff
@@ -72,16 +76,19 @@ public ModuleDeploymentId deploy(ModuleDeploymentRequest request) {
 		Map<String, String> rawArgs = new HashMap<>();
 		rawArgs.putAll(request.getDefinition().getParameters());
 		rawArgs.putAll(request.getDeploymentProperties());
-		Map<String, String> qualifiedArgs = ModuleArgumentQualifier.qualifyArgs(0, rawArgs);
+		int index = 0;
+		Map<String, String> qualifiedArgs = ModuleArgumentQualifier.qualifyArgs(index, rawArgs);
 		for (Map.Entry<String, String> entry : qualifiedArgs.entrySet()) {
 			environmentVariables.add(new EnvironmentVariable(entry.getKey(), entry.getValue()));
 		}
 		lrp.setEnv(environmentVariables.toArray(new EnvironmentVariable[environmentVariables.size()]));
 		lrp.setMemoryMb(512);
-		lrp.setPorts(new int[] {8080, 9000});
-
-		lrp.addHttpRoute(8080, new String[] {guid + "." + BASE_ADDRESS, guid + "-8080." + BASE_ADDRESS});
-		lrp.addHttpRoute(9000, new String[] {guid + "-9000." + BASE_ADDRESS});
+		String argsPrefix = String.format("%s.%s.", "args", index);
+		int serverPort = StringUtils.hasText(qualifiedArgs.get(argsPrefix + SERVER_PORT_KEY)) ?
```

Comment:
Can't you try to grab 'serverPort' from `rawArgs` above instead? Then there would be no need to mess with the prefix, and it would be similar to the local deployer: https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-module-deployers/spring-cloud-dataflow-module-deployer-local/src/main/java/org/springframework/cloud/dataflow/module/deployer/local/LocalModuleDeployer.java#L73